cpp/src/groupby/new_groupby.cu

```diff
@@ -350,6 +363,12 @@ gdf_group_by_without_aggregations(cudf::table const& input_table,
   std::transform(key_col_indices, key_col_indices+num_key_cols, key_cols_vect_out.begin(),
                   [&destination_table] (gdf_index_type const index) { return destination_table.get_column(index); });
   cudf::table key_col_sorted_table(key_cols_vect_out.data(), key_cols_vect_out.size());
+
+  // Copy the indices from device vector to gdf_column
+  rmm::device_vector<gdf_index_type> tmp = gdf_unique_indices(key_col_sorted_table, *context);
+  RMM_TRY(RMM_ALLOC(&unique_indices.data, sizeof(gdf_index_type)*tmp.size(), nullptr));
+  CUDA_TRY(cudaMemcpy(unique_indices.data, thrust::raw_pointer_cast(&tmp.front()), sizeof(gdf_index_type)*tmp.size(), cudaMemcpyDeviceToDevice));
```

Comment:
This is inefficient. `gdf_unique_indices` should be updated to also return a `gdf_column`.